kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -267,7 +267,10 @@ public GroupCoordinator getGroupCoordinator() {
     }
 
     public TransactionCoordinator getTransactionCoordinator() {
-        return tenantContextManager.getTransactionCoordinator(getCurrentTenant());
+        if (kafkaConfig.isKafkaTransactionCoordinatorEnabled()) {
+            return tenantContextManager.getTransactionCoordinator(getCurrentTenant());
+        }
+        return null;
```

Comment:
It's better to throw an exception here rather than returning a `null` value. Otherwise, the caller would have to call `throwIfTransactionCoordinatorDisabled` each time before `getTransactionCoordinator` is called.

Reply:
Good idea.